internal/controller/controllers/clusterdeployments_controller.go

```diff
@@ -235,13 +240,181 @@ func (r *ClusterDeploymentsReconciler) Reconcile(origCtx context.Context, req ct
 		}
 		return r.installDay1(ctx, log, clusterDeployment, clusterInstall, cluster, pullSecret)
 	} else if swag.StringValue(cluster.Kind) == models.ClusterKindAddHostsCluster {
+		// Set HTTPS Ignition URL
+		mcsSecretKey, err := r.createMCSCertSecret(ctx, log, clusterDeployment, cluster, clusterInstall)
```

Comment:
Since this controller is the one creating this secret I think it should also add a finalizer that deletes the secret when the cluster deployment get deleted.